### ISSUE TITLE
Strict equality false negative with 'in' and partial type

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#8182]
Strict equality checking fails in this example because of a partial type:

```py
x = []
for s in 'x', 'y':
    if 1 in x:  # No error, but should be
        x.append(s)
```

See #8177 for context.

**Comment by ilevkivskyi** [comment]
This now gives a correct error:
```
test.py:3: error: Non-overlapping container check (element type: "int", container item type: "str")  [comparison-overlap]
```
